packages/mobile/src/paymentRequest/PaymentRequestConfirmation.tsx

```diff
@@ -112,72 +94,60 @@ class PaymentRequestConfirmation extends React.Component<Props> {
 
     const paymentInfo = {
       amount: amount.toString(),
-      comment: this.state.comment || undefined,
+      comment: comment || undefined,
       createdAt: firebase.database.ServerValue.TIMESTAMP,
-      requesterAddress: address,
-      requesterE164Number: this.props.e164PhoneNumber ?? undefined,
+      requesterAddress: account,
+      requesterE164Number: confirmationInput.recipient.e164PhoneNumber ?? undefined,
```

Comment:
I think this `requesterE164Number` is meant to be the user's phone number, not the recipient's. You can get it with `e164NumberSelector`.